gadgets/Makefile

```diff
@@ -88,22 +88,36 @@ clean:
 
 .PHONY:
 test-ig-local: build
-	IG_PATH=$(IG) \
+	@overall_success=0; \
+	export IG_PATH=$(IG) \
 	IG_RUNTIME=docker \
 	CGO_ENABLED=0 \
 	IG_EXPERIMENTAL=true \
 	GADGET_REPOSITORY=$(GADGET_REPOSITORY) \
 	GADGET_TAG=$(GADGET_TAG) \
-	IG_FLAGS=$(IG_FLAGS) \
-	find . -name '*_test.go' -execdir go test -exec 'sudo -E' -v ./... \;
+	IG_FLAGS=$(IG_FLAGS); \
+	for dir in $$(find . -name '*_test.go' -exec dirname {} \;); do \
+	  (cd $$dir && go test -exec 'sudo -E' -v ./...); \
+	  if [ $$? -ne 0 ]; then \
+	    overall_success=1; \
+	  fi; \
+	done; \
```

Comment:
Also, is there a way to run this on parallel? I added t.Parallel() to all tests but it didn't do anything as they're run sequentially here.

Reply:
I actually don't know. But if we parallize it we also need to limit the number of concurrent jobs.

Reply:
We could also move the wasm layer into a subfolder, so the go file and the godot files are not hindering our testruns we did with `go test`.

WDYT?

Reply:
It makes a lot of sense. I agree, it'd be better to have a wasm module with all wasm related files (go.mod, go.sum, program.go), and then we won't have this annoying issue of the test runs.